doc/wait-for-ready.md

```diff
@@ -1,14 +1,26 @@
 gRPC Wait for Ready Semantics
 =============================
 
-If an RPC is issued but the channel is in `TRANSIENT_FAILURE` or `SHUTDOWN`
-states, the RPC is unable to be transmited promptly. By default, gRPC
-implementations SHOULD fail such RPCs immediately. This is known as "fail fast,"
-but usage of the term is historical. RPCs SHOULD NOT fail as a result of the
-channel being in other states (`CONNECTING`, `READY`, or `IDLE`).
-
-gRPC implementations MAY provide a per-RPC option to not fail RPCs as a result
-of the channel being in `TRANSIENT_FAILURE` state. Instead, the implementation
-queues the RPCs until the channel is `READY`. This is known as "wait for ready."
-The RPCs SHOULD still fail before `READY` if there are unrelated reasons, such
-as the channel is `SHUTDOWN` or the RPC's deadline is reached.
+Errors can occur in many places. For the propose of this document we can bucket the errors into three separate situations:
+
+1. The RPC never leaves the gRPC client library, for example:
+    * The connection was broken and the channel is in a `TRANSIENT_FAILURE` state.
+2. The RPC reaches the gRPC server library, and is rejected. For example:
```

Comment:
Bringing this into things is re-defining, not "clarifying". I think this is mostly orthogonal to wait-for-ready/fail-fast. "fail fast" should be permitted to be replayed here. Now, we may choose not to, but I don't think we should permanently disallow it. We can discuss this particular case somewhere else; it doesn't need to be here as it has nothing to do with the semantics of either fail fast or wait-for-ready.

Reply:
To be discussed